modules/core/src/system.cpp

```diff
@@ -1834,7 +1834,7 @@ FLAGS getFlags()
 #endif
 }
 
-NodeData::NodeData(const char* funName, const char* fileName, int lineNum, void* retAddress, bool alwaysExpand, cv::instr::TYPE instrType, cv::instr::IMPL implType)
+NodeData::NodeData(const cv::String funName, const char* fileName, int lineNum, void* retAddress, bool alwaysExpand, cv::instr::TYPE instrType, cv::instr::IMPL implType)
 {
     m_funName       = funName;
```

Comment:
Instrumentation should have minimal performance impact.
`const char*` is enough in parameters (it designed for using of static const string literals only).
It is better to fix this "assignment" statement (`std::string` doesn't allow to pass NULL).
```
m_funName       = funName ? funName : "";
```